scripts/determine_tests_to_run.py

```diff
@@ -65,11 +67,25 @@ def get_changed_files(branch: str) -> Tuple[List[str], List[str]]:
         ["git", "diff", "HEAD", branch, "--name-only"], stdout=subprocess.PIPE
     )
     files = [f.decode("utf-8") for f in process.stdout.splitlines()]
-    test_files = [f for f in files if f.startswith("tests")]
-    source_files = [f for f in files if f.startswith("great_expectations")]
+    source_files = get_changed_source_files(files, source_path)
+    test_files = get_changed_test_files(files, tests_path)
     return source_files, test_files
 
 
+def get_changed_source_files(files: List[str], source_path: str) -> List[str]:
+    valid_src = lambda f: f.startswith(source_path) and f.endswith(".py")
+    return [f for f in files if valid_src(f)]
+
+
+def get_changed_test_files(files: List[str], tests_path: str) -> List[str]:
+    valid_test = (
+        lambda f: f.startswith(tests_path)
+        and f.endswith(".py")
+        and f.split("/")[-1].startswith("test_")
+    )
```

Comment:
Should probably be a regex but this ensures we don't get extraneous files that break everything (i.e. the Titanic YAMLs)